### PR TITLE
Added yaml.encode and a few test cases for it

### PR DESCRIPTION
**Pull request by scr-oath**

### CODE REVIEW

File: yaml/api.go [hunk 1]
```diff
@@ -2,6 +2,7 @@
 package yaml
 
 import (
+	"fmt"
```
Comment (vadv) [inline yaml/api.go:5]: please update via golang.org/x/tools/cmd/goimports

**Review by vadv — CHANGES_REQUESTED** [review]
thanks 🙏 
please upgrade readme && rebase with master

File: yaml/api.go (hunk 1 above)
Comment (scr-oath) [reply yaml/api.go:5]: Done

**Comment by scr-oath** [comment]
Updated readme

**Comment by scr-oath** [comment]
> thanks 🙏
> please upgrade readme && rebase with master

Done

**Review by vadv — APPROVED** [review]

**Pull request merged**